tests/Tests.HubSpot.Client/Pipelines/HubSpotPipelineClientTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿using AutoFixture;
+using AutoFixture.Idioms;
+using AutoFixture.NUnit3;
+using HubSpot;
+using HubSpot.Model.Pipelines;
+using Kralizek.Extensions.Http;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Tests.Pipelines
+{
+    [TestFixture]
+    public class HubSpotPipelineClientTests
+    {
+        [Test, CustomAutoData]
+        public void Constructor_is_guarded_against_null_parameters(GuardClauseAssertion guardClauseAssertion)
+        {
+            //Arrange
+
+            //Act
+
+            //Assert
+            guardClauseAssertion.Verify(typeof(HttpHubSpotClient).GetConstructors());
+        }
+
+        [Test, CustomAutoData]
+        public void GetByGuidAsync_throws_NotFoundException_if_request_returns_NotFound_status([Frozen] IHttpRestClient httpRestClient, IHubSpotPipelineClient sut, string guid)
+        {
+            //Arrange
+            Mock.Get(httpRestClient)
+                .Setup(p => p.SendAsync<Pipeline>(HttpMethod.Get, $"/deals/v1/pipelines/{guid}", null))
+                .Throws(new HttpException("Not Found", HttpStatusCode.NotFound));
+
+            //Act
```

Comment:
Instead of mocking the httpclient, we can use the WorldDomination package to "mock" the http client calls (see the other tests)